src/producer.rs

```diff
@@ -254,20 +257,19 @@ impl<P: Partitioner> Producer<P> {
               V: AsBytes
     {
         let mut rs = try!(self.send_all(ref_slice(rec)));
-        if self.config.required_acks == 0 {
+
+        if self.config.required_acks != 0 {
             // ~ with no required_acks we get no response and
             // consider the send-data request blindly as successful
-            Ok(())
-        } else {
             assert_eq!(1, rs.len());
-            rs.pop().unwrap().offset.map(|_| ())
+            rs.pop().unwrap();
         }
+
+        Ok(())
```

Comment:
Well, yes, it would make sense. However, I consider `Producer` to be a higher-level API exposing less of the underlying details, therefore my first reaction would rather be to make `Producer::send_all` return a `Result<()>` instead (however, this would first require proper error handling within the `Producer` implementation.)

For some time, I'm playing with the idea to introduce some (user configurable) buffering into `Producer` such that `Producer::send/send_all` would first build up batches of message sets and send the records eventually only when some condition is met (e.g. the buffer gets full and/or some timeout is hit.) The idea is to give clients an easy API to work with, while allowing to send messages in batches - if they arrive quickly enough - which of course is much more efficient than sending them one by one. Such a feature would require that the `send/send_all` methods more or less do not return any of the "partition or offset information" returned by the broker.

At this point in time I have no preference and I'm fine with `Producer::send` returning `Result<ProduceConfirm>` now. I think it will be consistent with the `send_all` method.